velox/functions/sparksql/registration/Register.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/registration/Register.h"
+
+#include "velox/expression/SimpleFunctionRegistry.h"
+#include "velox/expression/SpecialFormRegistry.h"
+
+namespace facebook::velox::functions::sparksql {
+
+extern void registerArithmeticFunctions(const std::string& prefix);
+extern void registerArrayFunctions(const std::string& prefix);
+extern void registerBitwiseFunctions(const std::string& prefix);
+extern void registerCompareFunctions(const std::string& prefix);
+extern void registerDatetimeFunctions(const std::string& prefix);
+extern void registerJsonFunctions(const std::string& prefix);
+extern void registerMapFunctions(const std::string& prefix);
+extern void registerMiscFunctions(const std::string& prefix);
+extern void registerSpecialFormGeneralFunctions(const std::string& prefix);
+extern void registerStringFunctions(const std::string& prefix);
```

Comment:
The documents for Spark sql functions are categorized, while there are some inconsistencies on the two ways of categorization, e.g. 'greatest' is documented as a compare function, while in the registration it is a misc function. Such inconsistency might bring confusion to the reader.

https://github.com/facebookincubator/velox/tree/main/velox/docs/functions/spark